MapboxCoreNavigation/EventDetails.swift

```diff
@@ -105,19 +104,15 @@ struct EventDetails: Encodable {
             estimatedDuration = nil
         }
         
-        distanceCompleted = round(session.totalDistanceCompleted + router.routeProgress.distanceTraveled)
-        distanceRemaining = round(router.routeProgress.distanceRemaining)
-        durationRemaining = round(router.routeProgress.durationRemaining)
+        distanceCompleted = round(session.totalDistanceCompleted + dataSource.routeProgress.distanceTraveled)
+        distanceRemaining = round(dataSource.routeProgress.distanceRemaining)
+        durationRemaining = round(dataSource.routeProgress.durationRemaining)
         
         rerouteCount = session.numberOfReroutes
         
-        if let manager = router.locationManager {
-            locationEngine = String(describing: manager)
-            locationManagerDesiredAccuracy = manager.desiredAccuracy
-        } else {
-            locationEngine = nil
-            locationManagerDesiredAccuracy = nil
-        }
+        locationEngine = dataSource.locationSource.description
```

Comment:
Use the name of the class instead. `LocationSource.description` only supports simulation or device but you can also simulate on a device, or replay using the simulator, or use a custom location source. This information is valuable when debugging events.